pkg/agent/controller/traceflow/packetin.go

```diff
@@ -249,18 +251,64 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*crdv1alpha1.Tracefl
 				return nil, nil, nil, err
 			}
 		}
+		var pktMark uint32
+		if match := getMatchPktMarkField(matchers); match != nil {
+			pktMark, err = getMarkValue(match)
+			if err != nil {
+				return nil, nil, nil, err
+			}
+		}
+		var isRemoteEgress uint32
+		if match := getMatchRegField(matchers, openflow.RemoteSNATRegMark.GetField()); match != nil {
+			isRemoteEgress, err = getRegValue(match, openflow.RemoteSNATRegMark.GetField().GetRange().ToNXRange())
+			if err != nil {
+				return nil, nil, nil, err
+			}
+		}
 		gatewayIP := c.nodeConfig.GatewayConfig.IPv4
 		if etherData.Ethertype == protocol.IPv6_MSG {
 			gatewayIP = c.nodeConfig.GatewayConfig.IPv6
 		}
 		gwPort := c.nodeConfig.GatewayConfig.OFPort
 		tunPort := c.nodeConfig.TunnelOFPort
 		if c.networkConfig.TrafficEncapMode.SupportsEncap() && outputPort == tunPort {
+			if isRemoteEgress == 1 { // an Egress packet, currently on Source Node and forwarded to Egress Node.
+				obEgress := new(crdv1alpha1.Observation)
+				obEgress.EgressIP = tunnelDstIP
+				egress, err := c.egressQuerier.GetEgress(ns, srcPod)
+				if err != nil {
+					return nil, nil, nil, err
+				}
```

Comment:
nit: this may be less redundant (for repeating the variable name "obEgress"):
```
egress, err := c.egressQuerier.GetEgress(ns, srcPod)
if err != nil {
    return nil, nil, nil, err
}
obEgress := crdv1alpha1.Observation{
    Egress: egress,
    EgressIP: tunnelDstIP,
    ...
}
obs = append(obs, obEgress)
```